### PR TITLE
added specific fields to return on get request

### PR DESCRIPTION
**Pull request by 8rianHo**
specified the exact fields returned from a GET users call

based off of plu-15 PR

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -8,8 +8,13 @@ const app = express();
 app.use(express.json());
 
 app.get("/users", async (req, res) => {
-  const users = await Users.find({});
-  res.status(200).send(users);
+  const fields = "first_name family_name job_title job_company city socials";
+  try {
+    const users = await Users.find({}, fields);
```
Comment (lujasmine) [inline app.js:13]: Would we need to add the is approved field as a filter here, if we're using this route for the UI? But would it make the route too specific?

File: app.js [hunk 2]
```diff
@@ -8,8 +8,13 @@ const app = express();
 app.use(express.json());
 
 app.get("/users", async (req, res) => {
-  const users = await Users.find({});
-  res.status(200).send(users);
+  const fields = "first_name family_name job_title job_company city socials";
```
Comment (lujasmine) [inline app.js:11]: Didn't know you could query as a string like this!

What's the reason for just selecting these fields - and not age for example? Or returning the whole user object and filtering in the client side?

File: app.js [hunk 3]
```diff
@@ -8,8 +8,13 @@ const app = express();
...
+  try {
+    const users = await Users.find({}, fields);
+    res.status(200).send(users);
+  } catch (e) {
+    res.status(404).send();
```
Comment (lujasmine) [inline app.js:16]: I think 404 would only be if the query returns no users? Is it possible that the query could fail e.g. disconnected from mongo, where we should return a 500 instead?

Comment (lujasmine) [reply app.js:16]: Thinking about this, I don't think 404 makes sense here? If there are no users, then it should just return `[]`? Could go for the generic 500 here, and log any error messages in the server

File: app.js (hunk 1 above)
Comment (8rianHo) [reply app.js:13]: oh yeah you're right! I will have a think ☁️ query param is probably the better option

File: app.js (hunk 2 above)
Comment (8rianHo) [reply app.js:11]: We could be exposing extra data that the UI doesn't use but is available to the public i.e. I could go into the network tab and just grab all the emails/age from the response. I guess it's just making sure we only return what we need?

File: app.js [hunk 4]
```diff
@@ -8,15 +8,20 @@ const app = express();
...
 
 app.get("/users", async (req, res) => {
-  const users = await Users.find({});
-  res.status(200).send(users);
+  const fields = "first_name family_name job_title job_company city socials";
+  try {
+    const users = await Users.find({}, fields);
+    res.status(200).send(users);
+  } catch (e) {
+    res.status(500).send();
```
Comment (lujasmine) [inline app.js:16]: I think `res.sendStatus(500)` works as well rather than sending nothing

File: app.js [hunk 5]
```diff
@@ -8,15 +8,20 @@ const app = express();
...
+  }
 });
 
 app.post("/users", async (req, res) => {
   const user = new Users(req.body);
   try {
     await user.save();
-    res.status(201).send("created new user");
+    res.status(201).send();
```
Comment (lujasmine) [inline app.js:24]: same here for `sendStatus(201)` and below

File: app.js (hunk 4 above)
Comment (8rianHo) [reply app.js:16]: thought I'd keep the response standardised but can use sendStatus too - I will update

File: app.js (hunk 5 above)
Comment (8rianHo) [reply app.js]: updated for 201 and 400

**Review by lujasmine — APPROVED** [review]

**Pull request merged**